### ISSUE TITLE
No method "predict_classes" in Model

### ISSUE THREAD

**Issue by phdowling** [fchollet/keras#4288]
It seems that only Sequential models have the `predict_classes` method, which means that a model built using the functional API cannot be wrapped using the SKlearn `KerasClassifier` class. Is there a technical reason this is not implemented?

**Comment by Manezx** [comment]
`

    def probas_to_classes(y_pred):
        if len(y_pred.shape) > 1 and y_pred.shape[1] > 1:
            return categorical_probas_to_classes(y_pred)
        return np.array([1 if p > 0.5 else 0 for p in y_pred])

`

> from keras.utils.np_utils import probas_to_classes

**Comment by simonm3** [comment]
I  just came across same issue. I thought the functional and sequential apis were just alternative apis with the same underlying model. Why can't sklearn predict work for both rather than calling a function only available to sequential?

And more widely are there any other pitfalls using the sklearn wrapper with the functional api. I note that the code has many references to Sequential but sees to work fine with the functional model in most cases.

**Comment by mongoose54** [comment]
@Manezx  As of Keras 2.x there is no from  `probas_to_classes` . Any alternative?

**Comment by Manezx** [comment]
Here is an example for nlp
`

    labels = list(set(y))
    nb_classes = len(labels) + 1
    label2ind = {label: index+1 for index, label in enumerate(labels)}
    ind2label = {(index+1): label for index, label in enumerate(labels)}

    def encode(y):
        return np.asarray([to_categorical([label2ind[word] for word in sentence], nb_classes) for  sentence in y])

    def decode(y):
        y_ind = np.argmax(y, axis=-1)
        return [[ind2label[word] for word in sentence] for sentence in y_ind]`

**Comment by spedygiorgio** [comment]
It would be nice to add this method to the Model class, in order to give the same funtionalities of the Sequential api
